upstairs/src/dummy_downstairs_tests.rs

```diff
@@ -457,29 +448,28 @@ impl TestHarness {
 
         let join_handle = up_main(crucible_opts, 1, None, io, None).unwrap();
 
-        let mut handles: Vec<JoinHandle<Result<()>>> = vec![];
+        let mut handles: Vec<JoinHandle<()>> = vec![];
 
         {
             let guest = guest.clone();
             handles.push(tokio::spawn(async move {
-                guest.activate().await?;
-                Ok(())
+                guest.activate().await.unwrap();
             }));
         }
 
         // Connect all 3x Downstairs
         let mut fut = futures::stream::FuturesOrdered::new();
         for mut ds in [ds1, ds2, ds3] {
             fut.push_back(async move {
-                ds.negotiate_start().await?;
-                ds.negotiate_step_extent_versions_please().await?;
-                Result::<_>::Ok(ds)
+                ds.negotiate_start().await;
+                ds.negotiate_step_extent_versions_please().await;
+                ds
             });
         }
 
-        let ds1 = fut.next().await.unwrap().unwrap();
-        let ds2 = fut.next().await.unwrap().unwrap();
-        let ds3 = fut.next().await.unwrap().unwrap();
+        let ds1 = fut.next().await.unwrap();
+        let ds2 = fut.next().await.unwrap();
+        let ds3 = fut.next().await.unwrap();
```

Comment:
If we take out the `new_` returning result, is it still obvious from the stack which
of these ones fails?  (assuming I'm correct in that the change to `new_` is called
here.

Reply:
It looks like the `FuturesOrdered` will panic if any of the members panic, so the line number won't necessarily be meaningful.

It's debatable whether it matters, but c11aaf98b2b5d3500a9079133f622aa7441426a5 switches to checking the futures one by one, which _does_ give meaningful line numbers.